### PR TITLE
`Communication`: Improve the user interface

### PR DESCRIPTION
**Pull request by rabeatwork**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Tiny styling updates to improve the usability 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students

1. Log in to Artemis
2. Navigate to Communication
3. Verify that the gray backgrounds are removed 
4. Verify that the items in the sidebar are a bit smaller
5. Verify that messages and threads are working as before
- 

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Screenshot 2024-07-19 at 17 52 36
Screenshot 2024-07-19 at 18 04 11

## Summary by CodeRabbit

- **New Features**
  - Enhanced user interface for conversation components with improved layout and responsiveness.
  - Updated styling for buttons and headings, increasing readability and visual consistency.

- **Bug Fixes**
  - Streamlined logic for displaying answer post interactions, improving clarity and usability.

- **Documentation**
  - Improved semantic clarity and maintainability in the HTML structure of components.

- **Tests**
  - Adjusted test cases to reflect changes in HTML structure and enhance test accuracy.
  - Integrated mock services for profile support in component tests to improve testing scenarios.

### CODE REVIEW

**Review by pzdr7 — APPROVED** [review]
The code looks good :+1:

**Review by asliayk — DISMISSED** [review]
tested locally and works as expected

File: src/main/webapp/app/overview/course-conversations/course-conversations.component.scss [hunk 1]
```diff
@@ -15,15 +15,18 @@
...
-    border-color: var(--bs-card-border-color);
-    border-radius: 0.25rem;
+.scrollable-content {
+    --message-input-height-dev: 193px;
+    --message-input-height-prod: 177px;
     overflow-y: auto;
     overflow-x: hidden;
-    max-height: calc(100vh - var(--header-height) - var(--message-input-height));
+    max-height: calc(100vh - var(--header-height) - var(--message-input-height-prod));
+    &.content-height-dev {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/overview/course-conversations/course-conversations.component.scss:24]:
```suggestion
    max-height: calc(100vh - var(--header-height) - var(--message-input-height-prod));

    &.content-height-dev {
```
Codacy wants an empty line here

File: src/main/webapp/app/overview/course-conversations/course-conversations.component.scss [hunk 2]
```diff
@@ -15,15 +15,18 @@
...
+        max-height: calc(100vh - var(--header-height) - var(--message-input-height-dev));
+    }
+    @media (max-width: 768px) {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/overview/course-conversations/course-conversations.component.scss:27]:
```suggestion
    }

    @media (max-width: 768px) {
```

File: src/main/webapp/app/overview/course-conversations/layout/conversation-messages/conversation-messages.component.scss [hunk 3]
```diff
@@ -34,13 +34,20 @@
...
+    --message-input-height-prod: 171px;
+    --message-input-height-dev: 187px;
     --search-height: 52px;
     --channel-header-height: 52px;
 
     .posting-infinite-scroll-container {
-        max-height: calc(75vh - var(--message-input-height) - var(--search-height) - var(--channel-header-height));
+        max-height: calc(75vh - var(--message-input-height-prod) - var(--search-height) - var(--channel-header-height));
         overflow-y: auto;
+        &.content-height-dev {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/overview/course-conversations/layout/conversation-messages/conversation-messages.component.scss:45]:
```suggestion
        overflow-y: auto;

        &.content-height-dev {
```

File: src/main/webapp/app/overview/course-conversations/layout/conversation-messages/conversation-messages.component.scss [hunk 4]
```diff
@@ -34,13 +34,20 @@
...
+            max-height: calc(75vh - var(--message-input-height-dev) - var(--search-height) - var(--channel-header-height));
+        }
+        @media (max-width: 768px) {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/overview/course-conversations/layout/conversation-messages/conversation-messages.component.scss:48]:
```suggestion
        }

        @media (max-width: 768px) {
```

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Code looks good in general, but Coday complains about missing empty lines (see inline commments)

Additionally you touched code that uses the translate pipe. Can you update the lines to use jhiTranslate instead?

File: src/test/javascript/spec/component/overview/course-conversations/layout/conversation-header/conversation-header.component.spec.ts [hunk 5]
```diff
@@ -25,6 +25,10 @@ import { CourseExerciseDetailsComponent } from 'app/overview/exercise-details/co
 import { ExamDetailComponent } from 'app/exam/manage/exams/exam-detail.component';
 import { MetisService } from 'app/shared/metis/metis.service';
 import { MockMetisService } from '../../../../../helpers/mocks/service/mock-metis-service.service';
+import { MetisModule } from 'app/shared/metis/metis.module';
+import { MockTranslateService } from '../../../../../helpers/mocks/service/mock-translate.service';
+import { TranslateService } from '@ngx-translate/core';
+
```
Comment (pzdr7) [inline src/test/javascript/spec/component/overview/course-conversations/layout/conversation-header/conversation-header.component.spec.ts:32]:
```suggestion

```

File: src/test/javascript/spec/component/shared/metis/postings-header/answer-post-header/answer-post-header.component.spec.ts [hunk 6]
```diff
@@ -54,6 +55,7 @@ describe('AnswerPostHeaderComponent', () => {
                 MockComponent(PostingButtonComponent),
                 MockComponent(FaIconComponent),
                 MockComponent(ConfirmIconComponent),
+                // MockDirective(TranslateDirective),
```
Comment (pzdr7) [inline src/test/javascript/spec/component/shared/metis/postings-header/answer-post-header/answer-post-header.component.spec.ts:58]: Do we still need this?

**Review by JohannesWt — DISMISSED** [review]
Tested on TS4 - works as expected 👍

**Review by asliayk — APPROVED** [review]
reapprove

**Review by egekurt123 — COMMENTED** [review]
Works as expected on TS5, but i think we need to have some seperation between posts 
Screenshot 2024-07-26 at 11 32 05

**Review by egekurt123 — DISMISSED** [review]

File: src/test/javascript/spec/component/shared/metis/postings-header/answer-post-header/answer-post-header.component.spec.ts [hunk 7]
```diff
@@ -79,7 +83,7 @@ describe('AnswerPostHeaderComponent', () => {
 
     it('should set author information correctly', () => {
         fixture.detectChanges();
-        const headerAuthorAndDate = getElement(debugElement, '.header-author-date');
+        const headerAuthorAndDate = getElement(debugElement, '#header-author-date');
```
[the remark this answers is not in the document]
Comment (rabeatwork) [reply src/test/javascript/spec/component/shared/metis/postings-header/answer-post-header/answer-post-header.component.spec.ts:85]: it is.

File: src/test/javascript/spec/component/shared/metis/postings-header/answer-post-header/answer-post-header.component.spec.ts [hunk 8]
```diff
@@ -88,7 +92,7 @@ describe('AnswerPostHeaderComponent', () => {
         component.posting.creationDate = yesterday;
         component.ngOnInit();
         fixture.detectChanges();
-        expect(getElement(debugElement, '.today-flag')).toBeNull();
+        expect(getElement(debugElement, '#today-flag')).toBeNull();
```
[the remark this answers is not in the document]
Comment (rabeatwork) [reply src/test/javascript/spec/component/shared/metis/postings-header/answer-post-header/answer-post-header.component.spec.ts:94]: it is.

**Review by asliayk — APPROVED** [review]
reapprove

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by JohannesWt — APPROVED** [review]
Tested on TS4 - reaaprove

**Pull request merged**